Ticketing/Ticketing.DAL/Repositories/Repository.cs

```diff
@@ -22,39 +22,39 @@ public Repository(ApplicationContext context)
             _dbSet = _db.Set<T>();
         }
 
-        public async Task<T> CreateAsync(T entity)
+        public virtual async Task<T> CreateAsync(T entity)
         {
-            EntityEntry<T> addedEentity = await _dbSet.AddAsync(entity);
+            EntityEntry<T> addedEntity = await _dbSet.AddAsync(entity);
 
             await _db.SaveChangesAsync();
-            return addedEentity.Entity;
+            return addedEntity.Entity;
         }
 
-        public Task<IQueryable<T>> GetAllAsync()
+        public virtual Task<IQueryable<T>> GetAllAsync()
         {
             return Task.FromResult(_dbSet.AsQueryable());
```

Comment:
you do not need to return Task of IQueryable - as IQueryable is not a result, it is a query.
Theafore you do not need to use task from result.

Reply:
done